libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -578,7 +578,7 @@ void Compass::init()
 
 //  Register a new compass instance
 //
-int8_t Compass::register_compass(uint32_t dev_id)
+int8_t Compass::register_compass(int32_t dev_id)
```

Comment:
This change is in the wrong commit (`AP_Periph: use any_detected method for compass instances available or not`)

Reply:
Got it!